woocommerce/woocommerce-paypal-payments
# Fix Smart Buttons on Elementor checkout (1885)

### Description

This PR fixes the Smart Buttons on the Elementor Checkout by filtering the context.

It introduces two noteworthy changes:
- Adds a `woocommerce_paypal_payments_context` filter, to allow filtering of the context.
- Changes the order in which the compat module is being loaded, so that it can add a filter for the `woocommerce_paypal_payments_context`. 


### Steps to Test





1. Create a new page, and add the Elementor Checkout widget.
2. Make sure the Smart Button location `Checkout` is enabled.
3. Confirm that the Smart Buttons appear on the Elementor Checkout page correctly.


### Screenshots
|Before|After|
|-|-|
|![Elementor_Checkout_–_paypal](https://github.com/woocommerce/woocommerce-paypal-payments/assets/905781/8903cf44-bf5c-476b-a40b-24f46c79679a)|![Elementor_Checkout_–_paypal](https://github.com/woocommerce/woocommerce-paypal-payments/assets/905781/fd3a7951-0359-4f2b-84c9-f5e6706f7096)|
